IshiHisashi/GlucoFit
# feat: JDEV-62: Integrate backend to glucose log screen

## Description of this change
- add necessary libraries (apollo client, datetimepicker)
- create glucose log screen roughly based on wireframes
- add code to call GraphQL API to post a log
- add page navigation after posting a log
- styling and bottom sheet animation are NOT prioritized. 

## Steps to test
1. `cd frontend`, `npm install`, and start the app on an emulator
2. navigate yourself to the glucose log screen (center action button on the nav -> Blood Glucose)
3. input glucose level, pick date, time, time period, and input note
4. on save, you'll be navigated to home screen. You'll see the created log on the terminal or the DB. 

<img src="https://github.com/user-attachments/assets/a607a25f-4717-4e61-8889-ec0d17b67060" width="150px" />
<img src="https://github.com/user-attachments/assets/bb5957e7-dfaf-4951-ab94-4ee9d0c6a677" width="150px" />
<img src="https://github.com/user-attachments/assets/819f19c8-4cc9-4a34-8d5b-d9ade5e14f6d" width="150px" />
<img src="https://github.com/user-attachments/assets/30b535e5-51e2-4181-a9b5-c4ee5219715a" width="150px" />
<img src="https://github.com/user-attachments/assets/d624b2df-97e1-4d5a-95e6-b98ffb0d0bec" width="150px" />
<img src="https://github.com/user-attachments/assets/55ecf088-a6e1-4868-9ee9-c4abf6264b0c" width="150px" />
<img width="406" alt="image" src="https://github.com/user-attachments/assets/92607a81-5d5a-4bc1-a62b-33a75d83a7a8">


## Ticket link
https://team-jupiter-capstone.atlassian.net/browse/JDEV-62
